### PR TITLE
Use unix.Uname instead of shelling out to uname on darwin/freebsd

### PR DESCRIPTION
**Pull request by tklauser**
**- What I did**

Reuse the linux implementation based on `Uname` from `golang.org/x/sys/unix` to implement `GetOperatingSystem` in `pkg/parsers/operatingsystem` and `runtimeArchitecture` in `pkg/platform on darwin and freebsd.

**- How I did it**

See above.

**- How to verify it**

The functions return the same value as `uname -m` and `uname -s` on darwin/freebsd.

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

Gentoo penguin

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
wondering we can just remove darwin/freebsd code?

File: pkg/platform/architecture_unix.go [hunk 1]
```diff
@@ -1,20 +1,20 @@
-// +build freebsd darwin
+// +build freebsd darwin linux
```
Comment (AkihiroSuda) [inline pkg/platform/architecture_unix.go]: `// +build !windows`

**Comment by thaJeztah** [comment]
> wondering we can just remove darwin/freebsd code?

I think some code there was contributed, so I _suspect_ there might be people using it on those platforms

(PR https://github.com/moby/moby/pull/22474, ticket: https://github.com/moby/moby/issues/22470)

I'm fine with keeping it for now (doesn't look like there's a lot of maintenance involved in that package)

**Review by thaJeztah — APPROVED** [review]
LGTM after @AkihiroSuda's suggestion https://github.com/moby/moby/pull/39849#discussion_r320183803 is addressed

thanks again!

File: pkg/platform/architecture_unix.go (hunk 1 above)
Comment (tklauser) [reply pkg/platform/architecture_unix.go]: Done, thanks.

**Comment by thaJeztah** [comment]
ping @AkihiroSuda PTAL

**Review by AkihiroSuda — APPROVED** [review]

**Comment by thaJeztah** [comment]
Failure on Windows RS1; https://ci.docker.com/public/blue/rest/organizations/jenkins/pipelines/moby/branches/PR-39849/runs/5/nodes/50/log/?start=0

Looking at that test, I see it's taking a long time (491.77s, which is more than 8 minutes), so perhaps it's just too slow on RS1

Compared to RS5;

```
--- PASS: Test/DockerSuite/TestAPIImagesSaveAndLoad (146.25s)
```


```
 --- FAIL: Test/DockerSuite/TestAPIImagesSaveAndLoad (491.77s)
     cli.go:45: assertion failed: 
         Command:  d:\CI-5\CI-93d2cf881\binary\docker.exe inspect --format {{.Id}} sha256:69e7c1ff23be5648c494294a3808c0ea3f78616fad67bfe3b10d3a7e2be5ff02
         ExitCode: 1
         Error:    exit status 1
         Stdout:   
         
         Stderr:   Error: No such object: sha256:69e7c1ff23be5648c494294a3808c0ea3f78616fad67bfe3b10d3a7e2be5ff02
         
         
         Failures:
         ExitCode was 1 expected 0
         Expected no error
```

I'll open an issue for that test, and a PR to skip it on RS1 for now

**Comment by thaJeztah** [comment]
everything else is green, so merging this one; thanks!

**Pull request merged**